src/main/java/net/minecraftforge/common/data/SpriteSourceProvider.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.data;
+
+import com.mojang.serialization.JsonOps;
+import net.minecraft.client.renderer.texture.atlas.SpriteSource;
+import net.minecraft.client.renderer.texture.atlas.SpriteSources;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.PackType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+public abstract class SpriteSourceProvider extends JsonCodecProvider<List<SpriteSource>>
```

Comment:
A javadoc might be useful here, to explain what a sprite source does, or at least link relevant classes